In [1]:
import pandas as pd

In [2]:
dataset = pd.read_csv("Placement.csv")

In [3]:
dataset

,sl_no,gender,ssc_p,ssc_b,hsc_p,hsc_b,hsc_s,degree_p,degree_t,workex,etest_p,specialisation,mba_p,status,salary
0,1,M,67.00,Others,91.00,Others,Commerce,58.00,Sci&Tech,No,55.0,Mkt&HR,58.80,Placed,270000.0
1,2,M,79.33,Central,78.33,Others,Science,77.48,Sci&Tech,Yes,86.5,Mkt&Fin,66.28,Placed,200000.0
2,3,M,65.00,Central,68.00,Central,Arts,64.00,Comm&Mgmt,No,75.0,Mkt&Fin,57.80,Placed,250000.0
3,4,M,56.00,Central,52.00,Central,Science,52.00,Sci&Tech,No,66.0,Mkt&HR,59.43,Not Placed,NaN
4,5,M,85.80,Central,73.60,Central,Commerce,73.30,Comm&Mgmt,No,96.8,Mkt&Fin,55.50,Placed,425000.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
210,211,M,80.60,Others,82.00,Others,Commerce,77.60,Comm&Mgmt,No,91.0,Mkt&Fin,74.49,Placed,400000.0
211,212,M,58.00,Others,60.00,Others,Science,72.00,Sci&Tech,No,74.0,Mkt&Fin,53.62,Placed,275000.0
212,213,M,67.00,Others,67.00,Others,Commerce,73.00,Comm&Mgmt,Yes,59.0,Mkt&Fin,69.72,Placed,295000.0
213,214,F,74.00,Others,66.00,Others,Commerce,58.00,Comm&Mgmt,No,70.0,Mkt&HR,60.23,Placed,204000.0


In [4]:
from univariate import Univariate

In [5]:
Quan,Qual=Univariate.QuanQual(dataset)

In [6]:
Quan

['sl_no', 'ssc_p', 'hsc_p', 'degree_p', 'etest_p', 'mba_p', 'salary']

In [15]:
descriptive = pd.DataFrame(index=["Mean","Median","Mode","Q1:25 %","Q2:50 %","Q3:75 %","99 %","Q4:100 %"], columns=Quan)

In [16]:
descriptive

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
Mean,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Median,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Mode,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Q1:25 %,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Q2:50 %,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Q3:75 %,NaN,NaN,NaN,NaN,NaN,NaN,NaN
99 %,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Q4:100 %,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [10]:
import warnings
warnings.filterwarnings("ignore")

In [11]:
import numpy as np

In [12]:
np.percentile(dataset["ssc_p"],100)

np.float64(89.4)

In [17]:
for col in Quan:
    descriptive[col]["Mean"]=dataset[col].mean()
    descriptive[col]["Median"]=dataset[col].median()
    descriptive[col]["Mode"]=dataset[col].mode()[0]
#    descriptive[col]["Q1:25 %"] = np.percentile(dataset[col],25)
    descriptive[col]["Q1:25 %"] = dataset.describe()[col]["25%"]
#    descriptive[col]["Q2:50 %"] = np.percentile(dataset[col],50)
    descriptive[col]["Q2:50 %"] = dataset.describe()[col]["50%"]    
#    descriptive[col]["Q3:75 %"] = np.percentile(dataset[col],75)
    descriptive[col]["Q3:75 %"] = dataset.describe()[col]["75%"]    
#    descriptive[col]["Q4:100 %"] = np.percentile(dataset[col],100)
    descriptive[col]["99 %"] = np.percentile(dataset[col],99)
    descriptive[col]["Q4:100 %"] = dataset.describe()[col]["max"]


In [18]:
descriptive

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
Mean,108.0,67.303395,66.333163,66.370186,72.100558,62.278186,288655.405405
Median,108.0,67.0,65.0,66.0,71.0,62.0,265000.0
Mode,1,62.0,63.0,65.0,60.0,56.7,300000.0
Q1:25 %,54.5,60.6,60.9,61.0,60.0,57.945,240000.0
Q2:50 %,108.0,67.0,65.0,66.0,71.0,62.0,265000.0
Q3:75 %,161.5,75.7,73.0,72.0,83.5,66.255,300000.0
99 %,212.86,87.0,91.86,83.86,97.0,76.1142,NaN
Q4:100 %,215.0,89.4,97.7,91.0,98.0,77.89,940000.0


# Percentile Summary
For column: ssc_p
Q1:25% ssc_p: 60.6. In the data population 25% (1st quadrant) percentile data would be under 60.6.
Q2:50% ssc_p: 67.0. In the data population 50% (2nd quadrant) percentile data would be under 67.0. The value increaes from 1st Quadarand is 6.4.
Q3:75% ssc_p: 75.7. In the data population 75% (3rd quadrant) percentile data would be under 75.7. The value increaes from 2nd Quadarand is 8.7.
99 % : 87.0. In the data population 99%  percentile data would be under 87.0. The value increaes from 3rd Quadarand is 11.3.
Q4:100% ssc_p: 89.4. In the data population 100% (max) percentile data would be under 89.4. The value increaes from 3rd Quadarand is 13.7.

For column: hsc_p
Q1:25% hsc_p: 60.9. In the data population 25% (1st quadrant) percentile data would be under 60.9.
Q2:50% hsc_p: 65.0. In the data population 50% (2nd quadrant) percentile data would be under 65.0. The value increaes from 1st Quadarand is 4.1.
Q3:75% hsc_p: 73.0. In the data population 75% (3rd quadrant) percentile data would be under 73.0. The value increaes from 2nd Quadarand is 8.
99 % : 91.86. In the data population 99%  percentile data would be under 91.86. The value increaes from 3rd Quadarand is 18.86.
Q4:100% ssc_p: 97.7. In the data population 100% (max) percentile data would be under 97.7. The value increaes from 3rd Quadarand is 24.7.

For column: degree_p
Q1:25% degree_p: 61.0 In the data population 25% (1st quadrant) percentile data would be under 61.
Q2:50% degree_p: 66.0. In the data population 50% (2nd quadrant) percentile data would be under 66.0. The value increaes from 1st Quadarand is 5.
Q3:75% degree_p: 72.0. In the data population 75% (3rd quadrant) percentile data would be under 72.0. The value increaes from 2nd Quadarand is 6.
99 % : 83.86. In the data population 99%  percentile data would be under 83.86. The value increaes from 3rd Quadarand is 11.86.
Q4:100% ssc_p: 91 In the data population 100% (max) percentile data would be under 91. The value increaes from 3rd Quadarand is 19.

For column: etest_p
Q1:25% etest_p: 60.0. In the data population 25% (1st quadrant) percentile data would be under 60.0.
Q2:50% etest_p: 71.0. In the data population 50% (2nd quadrant) percentile data would be under 71.0. The value increaes from 1st Quadarand is 11.
Q3:75% etest_p: 83.5. In the data population 75% (3rd quadrant) percentile data would be under 83.5. The value increaes from 2nd Quad is 12.5.
99 % : 97. In the data population 99%  percentile data would be under 97. The value increaes from 3rd Quadarand is 28.
Q4:100% ssc_p: 98. In the data population 100% (max) percentile data would be under 98. The value increaes from 3rd Quadarand is 27.

For column: mba_p
Q1:25% mba_p: 57.9. In the data population 25% (1st quadrant) percentile data would be under 57.9.
Q2:50% mba_p: 62.0. In the data population 50% (2nd quadrant) percentile data would be under 62.0. The value increaes from 1st Quadarand is 4.1.
Q3:75% mba_p: 66.255. In the data population 75% (3rd quadrant) percentile data would be under 66.255. The value increaes from 2nd Quad is 4.255.
99 % :76.1. In the data population 99%  percentile data would be under 76.1. The value increaes from 3rd Quadarand is 9.9.
Q4:100% ssc_p: 77.89. In the data population 100% (max) percentile data would be under 77.89. The value increaes from 3rd Quadarand is 11.7.

In [21]:
descriptive = pd.DataFrame(index=["Mean","Median","Mode","Q1:25 %","Q2:50 %","Q3:75 %","99 %","Q4:100 %","IQR","1.5Rule","Lesser","Greater","Min","Max"], columns=Quan)
for col in Quan:
    descriptive[col]["Mean"]=dataset[col].mean()
    descriptive[col]["Median"]=dataset[col].median()
    descriptive[col]["Mode"]=dataset[col].mode()[0]
    descriptive[col]["Q1:25 %"] = dataset.describe()[col]["25%"]
    descriptive[col]["Q2:50 %"] = dataset.describe()[col]["50%"]    
    descriptive[col]["Q3:75 %"] = dataset.describe()[col]["75%"]    
    descriptive[col]["99 %"] = np.percentile(dataset[col],99)
    descriptive[col]["Q4:100 %"] = dataset.describe()[col]["max"]
    descriptive[col]["IQR"] = dataset.describe()[col]["75%"] -  dataset.describe()[col]["25%"]
    descriptive[col]["1.5Rule"] =1.5*(dataset.describe()[col]["75%"] -  dataset.describe()[col]["25%"])
    descriptive[col]["Lesser"] =dataset.describe()[col]["25%"] - (1.5*(dataset.describe()[col]["75%"] -  dataset.describe()[col]["25%"]))
    descriptive[col]["Greater"] =dataset.describe()[col]["75%"] + (1.5*(dataset.describe()[col]["75%"] -  dataset.describe()[col]["25%"]))
    descriptive[col]["Min"] = dataset[col].min()
    descriptive[col]["Max"] = dataset[col].max()

In [22]:
descriptive

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
Mean,108.0,67.303395,66.333163,66.370186,72.100558,62.278186,288655.405405
Median,108.0,67.0,65.0,66.0,71.0,62.0,265000.0
Mode,1,62.0,63.0,65.0,60.0,56.7,300000.0
Q1:25 %,54.5,60.6,60.9,61.0,60.0,57.945,240000.0
Q2:50 %,108.0,67.0,65.0,66.0,71.0,62.0,265000.0
Q3:75 %,161.5,75.7,73.0,72.0,83.5,66.255,300000.0
99 %,212.86,87.0,91.86,83.86,97.0,76.1142,NaN
Q4:100 %,215.0,89.4,97.7,91.0,98.0,77.89,940000.0
IQR,107.0,15.1,12.1,11.0,23.5,8.31,60000.0
1.5Rule,160.5,22.65,18.15,16.5,35.25,12.465,90000.0


In [24]:
if descriptive["hsc_p"]["Lesser"] < descriptive["hsc_p"]["Min"] :
    print("Not Outlier")
else:
    print("Outlier")

Outlier


In [29]:
lesser=[]
greater=[]
for cols in Quan:
    if descriptive[cols]["Min"]<descriptive[cols]["Lesser"]:
        lesser.append(cols)
    if descriptive[cols]["Max"]>descriptive[cols]["Greater"]:
        greater.append(cols)

print(lesser)
print(greater)

['hsc_p']
['hsc_p', 'degree_p', 'salary']


In [32]:
for columnName in lesser:
    dataset[columnName][dataset[columnName]<descriptive[columnName]["Lesser"]] = descriptive[columnName]["Lesser"]

for columnName in greater:
    dataset[columnName][dataset[columnName]>descriptive[columnName]["Greater"]] = descriptive[columnName]["Greater"]

In [33]:
descriptive = pd.DataFrame(index=["Mean","Median","Mode","Q1:25 %","Q2:50 %","Q3:75 %","99 %","Q4:100 %","IQR","1.5Rule","Lesser","Greater","Min","Max"], columns=Quan)
for col in Quan:
    descriptive[col]["Mean"]=dataset[col].mean()
    descriptive[col]["Median"]=dataset[col].median()
    descriptive[col]["Mode"]=dataset[col].mode()[0]
    descriptive[col]["Q1:25 %"] = dataset.describe()[col]["25%"]
    descriptive[col]["Q2:50 %"] = dataset.describe()[col]["50%"]    
    descriptive[col]["Q3:75 %"] = dataset.describe()[col]["75%"]    
    descriptive[col]["99 %"] = np.percentile(dataset[col],99)
    descriptive[col]["Q4:100 %"] = dataset.describe()[col]["max"]
    descriptive[col]["IQR"] = dataset.describe()[col]["75%"] -  dataset.describe()[col]["25%"]
    descriptive[col]["1.5Rule"] =1.5*(dataset.describe()[col]["75%"] -  dataset.describe()[col]["25%"])
    descriptive[col]["Lesser"] =dataset.describe()[col]["25%"] - (1.5*(dataset.describe()[col]["75%"] -  dataset.describe()[col]["25%"]))
    descriptive[col]["Greater"] =dataset.describe()[col]["75%"] + (1.5*(dataset.describe()[col]["75%"] -  dataset.describe()[col]["25%"]))
    descriptive[col]["Min"] = dataset[col].min()
    descriptive[col]["Max"] = dataset[col].max()

In [34]:
descriptive

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
Mean,108.0,67.303395,66.334744,66.358558,72.100558,62.278186,277648.648649
Median,108.0,67.0,65.0,66.0,71.0,62.0,265000.0
Mode,1,62.0,63.0,65.0,60.0,56.7,300000.0
Q1:25 %,54.5,60.6,60.9,61.0,60.0,57.945,240000.0
Q2:50 %,108.0,67.0,65.0,66.0,71.0,62.0,265000.0
Q3:75 %,161.5,75.7,73.0,72.0,83.5,66.255,300000.0
99 %,212.86,87.0,91.129,83.86,97.0,76.1142,NaN
Q4:100 %,215.0,89.4,91.15,88.5,98.0,77.89,390000.0
IQR,107.0,15.1,12.1,11.0,23.5,8.31,60000.0
1.5Rule,160.5,22.65,18.15,16.5,35.25,12.465,90000.0


In [35]:
lesser=[]
greater=[]
for cols in Quan:
    if descriptive[cols]["Min"]<descriptive[cols]["Lesser"]:
        lesser.append(cols)
    if descriptive[cols]["Max"]>descriptive[cols]["Greater"]:
        greater.append(cols)

print(lesser)
print(greater)

[]
[]
